### ISSUE TITLE
Understand exactly what the library can and cannot do

### ISSUE THREAD

**Issue by 8ctopus** [damianobarbati/get-browser-fingerprint#29]
Hello Damiano,

I've just discovered your library and I have taken it for a test run. Here's my results:

Edge Version 119.0.2151.58 (Official build) (64-bit) in normal mode

```txt
Fingerprint: 2803895700 (computed in 65 ms)
Fingerprint with hardwareOnly=true: 1801653769 (computed in 12 ms)
Fingerprint with enableWebgl=true: 1093115891 (computed in 288 ms)
```

Edge in private browsing

```txt
Fingerprint: 2352988901 (computed in 32 ms)
Fingerprint with hardwareOnly=true: 1801653769 (computed in 4 ms)
Fingerprint with enableWebgl=true: 1140287578 (computed in 80 ms)
```

Firefox 119.0.1 (64 bit) normal browsing

```txt
Fingerprint: 102973731 (computed in 101 ms)
Fingerprint with hardwareOnly=true: 2045863812 (computed in 9 ms)
Fingerprint with enableWebgl=true: 3031478298 (computed in 159 ms)
```

Firefox, clear all cache and normal browsing

```txt
Fingerprint: 102973731 (computed in 22 ms)
Fingerprint with hardwareOnly=true: 2045863812 (computed in 4 ms)
Fingerprint with enableWebgl=true: 3031478298 (computed in 58 ms)
```

Firefox private browsing

```txt
Fingerprint: 3324419382 (computed in 68 ms)
Fingerprint with hardwareOnly=true: 2045863812 (computed in 10 ms)
Fingerprint with enableWebgl=true: 1721380340 (computed in 114 ms)
```

Chromium Version 121.0.6100.0 (Developer Build) (64-bit) normal browsing

```txt
Fingerprint: 1776080590 (computed in 57 ms)
Fingerprint with hardwareOnly=true: 2109763731 (computed in 30 ms)
Fingerprint with enableWebgl=true: 3326289934 (computed in 188 ms)
```

after clearing the whole cache, normal browsing

```txt
Fingerprint: 1776080590 (computed in 87 ms)
Fingerprint with hardwareOnly=true: 2109763731 (computed in 10 ms)
Fingerprint with enableWebgl=true: 3326289934 (computed in 165 ms)
```

and private browsing

```txt
Fingerprint: 2246039315 (computed in 91 ms)
Fingerprint with hardwareOnly=true: 2109763731 (computed in 9 ms)
Fingerprint with enableWebgl=true: 2774134400 (computed in 149 ms)
```

So based on these results, I postulate that:
- every browser has its own fingerprints;
- all fingerprints remain the same after clearing the browser cache;
- the hardware only fingerprint is unaltered in private mode, all others change.

Is that correct? Are there any other findings?

**Comment by 8ctopus** [comment]
I should have said that I planning to use the library to detect fraudsters that are trying to test stolen credit cards on the payment page of one of my clients.

Besides your library, I found fingerprint dot com but their pricing starts at USD400 per month which I find outrageous.

**Comment by damianobarbati** [comment]
@8ctopus thanks for sharing this, all of your assumptions look correct to me:

> every browser has its own fingerprints
> all fingerprints remain the same after clearing the browser cache

Yes, the fingerprint identifies uniquely the browser.
If your goal is to identity the device (any browser, given your use cases) then you can try removing the `userAgent` [here](<https://github.com/damianobarbati/get-browser-fingerprint/blob/main/src/index.js#L63>): I'm afraid several bits may change anyway in the several factors considered when computing the fingerprint causing a different hash for each browser (I didn't test this, but give it a try and let me know!).

> the hardware only fingerprint is unaltered in private mode, all others change.
Also true, I can't remember now which factors are impacted in the private mode where vendors are try to prevent fingerprinting techniques.

I agree pricing is crazy, and also the most popular one I tried to use once was 1) async and 2) 500kb zipped heavy.

**Comment by 8ctopus** [comment]
Thank you for your feedback Damiano. I ended up doing something different to track people across browsers, since they keep using the same patterns for card testing, it was not so hard to follow the trail.
